kubernetes/client/kubernetes.go

```diff
@@ -0,0 +1,738 @@
+								},
+							},
+							Env: []apiv1.EnvVar{
+								{
+									Name:  "WR_MANAGERPORT",
+									Value: strconv.Itoa(requiredPorts[0]),
+								},
+								{
+									Name:  "WR_MANAGERWEB",
+									Value: strconv.Itoa(requiredPorts[1]),
+								},
+								{
+									Name:  "HOME",
+									Value: tempMountPath,
+								},
+							},
+							SecurityContext: &apiv1.SecurityContext{
+								Privileged: boolPtr(true),
+							},
+							Lifecycle: &apiv1.Lifecycle{
+								PreStop: &apiv1.Handler{
+									Exec: &apiv1.ExecAction{
+										Command: []string{"cat", tempMountPath + ".wr_development/log",
+											"cat", tempMountPath + ".wr_development/kubeSchedulerLog",
+											"cat", tempMountPath + ".wr_development/kubeSchedulerControllerLog"},
+									},
+								},
+							},
+						},
+					},
+					InitContainers: []apiv1.Container{
+						{
+							Name:      "init-container",
+							Image:     "ubuntu:17.10",
+							Command:   []string{"/bin/tar", "-xf", "-"},
+							Stdin:     true,
+							StdinOnce: true,
+							VolumeMounts: []apiv1.VolumeMount{
+								{
+									Name:      "wr-temp",
+									MountPath: tempMountPath,
+								},
+							},
+						},
+					},
+					Hostname: "wr-manager",
+				},
+			},
+		},
+	}
+
+	// Create Deployment
+	p.Logger.Info(fmt.Sprintf("Creating deployment..."))
+	result, err := p.deploymentsClient.Create(deployment)
+	if err != nil {
+		p.Logger.Error("creating deployment", "err", err)
+		return err
+	}
+	p.Logger.Info(fmt.Sprintf("Created deployment %q in namespace %v.\n", result.GetObjectMeta().GetName(), p.NewNamespaceName))
+	// specify service so wr-manager can be resolved using kubedns if needed
+	svcOpts := ServiceOpts{
+		Name: "wr-manager",
+		Labels: map[string]string{
+			"app": "wr-manager",
+		},
+		Selector: map[string]string{
+			"app": "wr-manager",
+		},
+		ClusterIP: "None",
+		Ports: []apiv1.ServicePort{
+			{
+				Name: "wr-manager",
+				Port: int32(requiredPorts[0]),
+				TargetPort: intstr.IntOrString{
+					IntVal: int32(requiredPorts[0]),
+				},
+			},
+		},
+	}
+	err = p.CreateService(&svcOpts)
+	if err != nil {
+		p.Logger.Error("creating service", "err", err)
+		return err
+	}
+
+	/*
+	   No longer copying tarball to pod here, instead the deployment controller
+	   waits on the status of the InitContainer to be running, then runs
+	   CopyTar, found in pod.go
+
+	*/
+
+	return nil
+}
+
+// InWRPod() checks if we are in a WR pod. As we control the hostname, just
+// check if the hostname contains 'wr' in addition to the standard environment
+// variables.
+func InWRPod() bool {
+	hostname, err := os.Hostname()
+	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
+	inPod := false
+	if err == nil && len(host) != 0 && len(port) != 0 {
+		if strings.Contains(hostname, "wr") {
+			inPod = true
+		}
+	}
+	return inPod
+}
+
+// Spawn a new pod that contains a runner. Return the name.
```

Comment:
I don't think there's anything wr-specific about this method. Remove reference to runner and provide a better explanation of what this does.

Reply:
(It does force the naming of the pod to `wr-runner-XXXX`)